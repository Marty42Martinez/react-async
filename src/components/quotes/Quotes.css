.Quotes {
  display: flex;
  justify-content: space-around;
  flex-wrap: wrap;
  align-items: center;

  li {
    width: 20vw;
    margin: 2em;
    padding: 2rem;

    background-color: #a9c4f5;

    text-align: center;

    list-style-type: none;
  }

  section {
    margin: auto;
    animation: insert 1.5s linear;
  }

  img {
    width: 10rem;
    margin: auto;
  }

  @keyframes insert {
    from {
      width: 0;
    }
    to {
      width: 100%;
    }
  }
}