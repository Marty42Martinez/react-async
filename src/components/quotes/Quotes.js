import React from 'react';
import PropTypes from 'prop-types';
import Quote from './Quote';

function Quotes({ quotes }) {
  const quoteList = quotes.map((quote, i) => {
    return (
      <li key={i}>
        <Quote quote={quote} />
      </li>
    );
  });

  return (
    <ul>
      {quoteList}
    </ul>
  );
}

Quotes.propTypes = {
  quotes: PropTypes.array.isRequired
};
