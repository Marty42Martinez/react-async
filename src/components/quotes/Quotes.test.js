import React from 'react';
import { shallow } from 'enzyme';
import Quotes from './Quotes';

describe('Quotes component', () => {
  it('renders a Quotes', () => {
    const quotes = [
      { img: 'test', character: 'Fry', quote: 'hello' },
      { img: 'test2', character: 'Bender', quote: 'go away' }
    ];

    const wrapper = shallow(<Quotes quotes={quotes} />);
    expect(wrapper).toMatchSnapshot();
  });
});
