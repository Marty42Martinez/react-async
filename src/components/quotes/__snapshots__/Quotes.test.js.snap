// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quotes component renders a Quotes 1`] = `
<ul>
  <li
    key="0"
  >
    <Quote
      quote={
        Object {
          "character": "Fry",
          "img": "test",
          "quote": "hello",
        }
      }
    />
  </li>
  <li
    key="1"
  >
    <Quote
      quote={
        Object {
          "character": "Bender",
          "img": "test2",
          "quote": "go away",
        }
      }
    />
  </li>
</ul>
`;
